src/builder.ts

```diff
@@ -0,0 +1,70 @@
+import { resolveURL, withQuery } from 'ufo'
+import { defu } from 'defu'
+import type { QueryObject } from 'ufo'
+import type { FetchOptions } from './fetch'
+import type { ResponseType, MappedType } from './types'
+import { $fetch } from './undici'
+
+export type ClientMethodHandler = <T = any, R extends ResponseType = 'json'>(
+  data?: RequestInit['body'] | Record<string, any>,
+  opts?: Omit<FetchOptions<R>, 'baseURL' | 'method'>
+) => Promise<MappedType<R, T>>
+
+export type ClientBuilder = {
+  [key: string]: ClientBuilder;
+  (...segmentsOrIds: (string | number)[]): ClientBuilder
+} & {
+  get: ClientMethodHandler
+  post: ClientMethodHandler
+  put: ClientMethodHandler
+  delete: ClientMethodHandler
+  patch: ClientMethodHandler
+}
+
+export function createClient<R extends ResponseType = 'json'> (
+  url: string,
```

Comment:
We can simplify interface to `createClient({ baseURL, ...options })`.

Reply:
Great idea. So you mean `/` as the default base URL, right?
Please take a look at the current implementation: https://github.com/unjs/ohmyfetch/pull/69/commits/79098d08b2a4dcd106f6d52c988ea7881abb1525